File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        ArrayList<Integer> arr = new ArrayList<>();
        int[] res = new int[k];
        for(int i = 0; i < nums.length; i++){
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
        }
        List<Map.Entry<Integer, Integer>> list = new ArrayList<>(map.entrySet());
        list.sort((a, b) -> b.getValue().compareTo(a.getValue()));
        Map<Integer, Integer> sortedMap = new LinkedHashMap<>();
        for (Map.Entry<Integer, Integer> entry : list) {
            sortedMap.put(entry.getKey(), entry.getValue());
        }
        int count = 0;

        for(int c : sortedMap.keySet()){
            arr.add(c);
            count++;
            if(count == k){
                break;
            }

        }
        for(int j = 0; j < arr.size(); j++){
            res[j] = arr.get(j);
        }
        return res;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        prefix = [0] * n
        suffix = [0] * n
        res=[]
        prefix[0] = 1
        suffix[n-1] = 1
        for i in range(1, n):
            prefix[i] = prefix[i-1]*nums[i-1]

        for j in range(n-2, -1, -1):
            suffix[j] = suffix[j+1] * nums[j+1]

        for s in range(n):
            res.append(prefix[s]* suffix[s])
        
        return res